src/dune/build_system.mli

```diff
@@ -79,13 +81,13 @@ val prefix_rules : unit Build.t -> f:(unit -> 'a) -> 'a
 (** [eval_pred t [glob]] returns the list of files in [File_selector.dir glob]
     that matches [File_selector.predicate glob]. The list of files includes the
     list of targets. *)
-val eval_pred : File_selector.t -> Path.Set.t
+val eval_pred : File_selector.t -> Path.Set.t Fiber.t
 
 (** Returns the set of targets in the given directory. *)
-val targets_of : dir:Path.t -> Path.Set.t
+val targets_of : dir:Path.t -> Path.Set.t Fiber.t
 
 (** Load the rules for this directory. *)
-val load_dir : dir:Path.t -> unit
+val load_dir : dir:Path.t -> unit Fiber.t
```

Comment:
@diml have a look at the changes in this mli. As far as I can tell, the callback to gen_rules can call all the functions above so we are required to change the return type of the callback to `unit Fiber.t`.

Reply:
Indeed. That seems reasonable to me